### PR TITLE
[16.0][MIG] website_sale_product_attribute_value_filter_existing: Migration to 16.0

### PR DESCRIPTION
**Pull request by anothingguy**

### CODE REVIEW

File: website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/** @odoo-module **/
+
+import {session} from "@web/session";
+import tour from "web_tour.tour";
+
+var domReady = new Promise(function (resolve) {
+    $(resolve);
+});
+var ready = Promise.all([domReady, session.is_bound]);
```
Comment (chienandalu) [inline website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js]: I bet this isn't needed :)
```suggestion
```

File: website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+tour.register(
+    "website_sale_product_attribute_value_filter_existing",
+    {
+        test: true,
+        url: "/shop",
+        wait_for: ready,
```
Comment (chienandalu) [inline website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js]:
```suggestion
```

File: website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py [hunk 3]
```diff
@@ -71,13 +72,10 @@ def setUp(self):
...
+
+        self.start_tour(
+            "/",
             "website_sale_product_attribute_value_filter_existing",
-        )
-        self.browser_js(
-            url_path="/",
-            code="%s.run('%s')" % tour,
-            ready="%s.tours['%s'].ready" % tour,
+            step_delat=100,
```
Comment (chienandalu) [inline website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py]: Is it needed anyway?
```suggestion
            step_delay=100,
```

File: website_sale_product_attribute_value_filter_existing/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+* Go to Website Shop
+* Active product attributes filter
```
Comment (chienandalu) [inline website_sale_product_attribute_value_filter_existing/readme/USAGE.rst]: This has changed a bit in v16 as now is a theme panel option that is enabled on either top or bottom in the layout. I'd move it to CONFIGURATION.rst as well.

**Review by chienandalu — CHANGES_REQUESTED** [review]
Tested. Good work, thanks! :) Some minor comments:

File: website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js (hunk 1 above)
Comment (anothingguy) [reply website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js]: yep, at the beginning, I didn't add this ready, keep it same as v15. but without this ready, the running tests somehow is not stable, 50% passed and 50% failed. So I searched and tried to add this ready to make sure this tests run when DOM is ready

Comment (chienandalu) [reply website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js]: Yeah, when tours fail that randomly is quite annoying. Ok then :+1:

Comment (anothingguy) [reply website_sale_product_attribute_value_filter_existing/static/src/js/website_sale_product_attribute_value_filter_existing_tour.esm.js]: yeah, I removed this redundant code and just keep `step_delay` to make sure there is enough time between steps

File: website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py (hunk 3 above)
Comment (anothingguy) [reply website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py]: yes, I will keep this to make sure there is enough time between steps. Some tests in Odoo modules are still using this

File: website_sale_product_attribute_value_filter_existing/readme/USAGE.rst (hunk 4 above)
Comment (anothingguy) [reply website_sale_product_attribute_value_filter_existing/readme/USAGE.rst]: yes, moved to `CONFIGURATION.rst` and update more details. thanks for pointing out

**Review by chienandalu — APPROVED** [review]
Thanks! :+1:

**Review by StephaneMangin — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot migration website_sale_product_attribute_value_filter_existing

**Pull request merged**